_pages/oidc.md

```diff
@@ -133,12 +133,15 @@ https://idp.int.identitysandbox.gov/openid_connect/authorize?
   The unique identifier for the client. This will be registered with the Login.gov IdP in advance.
 
 * **code_challenge** — *required for PKCE*
-  The URL-safe base64 encoding of the SHA256 digest of a random value generated by the client. The original random value is referred to as the `code_verifier` and is later used with the token endpoint. Generating these values in Ruby might look like this, for example:
+  The URL-safe Base64 encoding of the SHA256 digest of a random value generated by the client. The original random value is referred to as the `code_verifier` and is later used with the token endpoint. Generating these values in Ruby might look like this, for example:
   ```ruby
   code_verifier = SecureRandom.hex
-  => "7a5e819dd39f17242fdeeba0c1c80be6"
+  => "5787d673fb784c90f0e309883241803d"
   code_challenge = Digest::SHA256.base64digest(code_verifier)
-  => "TdzfmaWefbtaI0Wdo6lrZCXpLu1WpamnSoSHfDUiL7Y="
+  => "1BUpxy37SoIPmKw96wbd6MDcvayOYm3ptT+zbe6L/zM="
+  url_safe_code_challenge = Base64.urlsafe_encode64(Base64.urlsafe_decode64(code_challenge))
```

Comment:
I feel like this example had a lot of hops. What if we updated it to do SHA256 digest as binary data and then use urlsafe base64 encoding of that?

Reply:
so more like:
```ruby
  code_challenge = Digest::SHA256.digest(code_verifier) # binary data
  code_challenge.bytes
  => [212, 21, 41, 199, 45, 251, 74, 130, 15, 152, 172, 61, 235, 6, 221, 232, 192, 220, 189, 172, 142, 98, 109, 233, 181, 63, 179, 109, 238, 139, 255, 51]  
  Base64.urlsafe_encode64(code_challenge)
  => "1BUpxy37SoIPmKw96wbd6MDcvayOYm3ptT-zbe6L_zM="
```

Reply:
For me, it was helpful to have the non-URL-safe encoding in the middle to see the differences between the steps.

Reply:
maybe we add a different "don't do this" example for the non-urlsafe at the end? I like the idea of having an easy-to-follow correct, happy path

```
Base64.encode64(code_challenge) # WRONG ENCODING
# => "1BUpxy37SoIPmKw96wbd6MDcvayOYm3ptT+zbe6L/zM=" # WRONG ENCODING
```

Reply:
Sounds good 👍🏼